### PR TITLE
Replace calls to Random([a..b]) by Random(a,b)

### PR DESCRIPTION
**Pull request by fingolfin**
This avoids some overhead for creating ranges

Inspired by PR #2930

### CODE REVIEW

**Comment by ChrisJefferson** [comment]
This probably doesn't matter, but just to mention, are you aware that Random([a..b]) and random(a,b) have different behaviour when b < a? Random([5..1]) enters a break loop, which Random(5,1) returns fail. I just mention it in case in some cases it could make a difference (obviously not whenever it's easy to see b >= a).

**Comment by fingolfin** [comment]
Yes, I am aware of it.

File: doc/ref/xtndxmpl.xml [hunk 1]
```diff
@@ -1329,7 +1329,7 @@ gap> InstallMethod( Random,
 >    "for full ring Z/nZ",
 >    [ CategoryCollections( IsMyZmodnZObjNonprime ) and IsWholeFamily ],
 >    R -> MyZmodnZObj( ElementsFamily( FamilyObj( R ) ),
->                    Random( [ 0 .. Size( R ) - 1 ] ) ) );
+>                    Random( 0 .. Size( R ) - 1 ) ) );
```
Comment (ChrisJefferson) [inline doc/ref/xtndxmpl.xml]: I guess this is a typo? (The .. ).

Comment (fingolfin) [reply doc/ref/xtndxmpl.xml]: Indeed, ouch :-(. I'll fix it, thank you very much for careful checking.

Comment (fingolfin) [reply doc/ref/xtndxmpl.xml]: Fixed

**Pull request merged**